### PR TITLE
Menu sz

### PR DESCRIPTION
**Pull request by RadekVana**
Calculate size of screen buffer automatically
BFW-3101

### CODE REVIEW

File: src/gui/screen_menu_info.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include "WindowMenuItems.hpp"
+#include "MItem_menus.hpp"
+
+using ScreenMenuInfo__ = ScreenMenu<EFooter::On, MI_RETURN,
+#ifdef _DEBUG
+    MI_FAIL_STAT_disabled, MI_SUPPORT_disabled,
+#endif //_DEBUG
+    MI_SYS_INFO, MI_SENSOR_INFO, MI_VERSION_INFO, MI_ODOMETER, MI_EEPROM_DIAGNOSTICS>;
+
+//cannot move it to header - 'ScreenMenuInfo' has a field 'ScreenMenuInfo::<anonymous>' whose type uses the anonymous namespace [-Wsubobject-linkage]
```
Comment (mkbel) [inline src/gui/screen_menu_info.hpp:16]: I don't understand this comment. Is it even still valid?

File: src/gui/screen_menu_sensor_info.cpp [hunk 2]
```diff
@@ -38,5 +19,5 @@ void ScreenMenuSensorInfo::windowEvent(EventLock /*has private ctor*/, window_t
         return;
     }
 
-    Screen::SuperWindowEvent(sender, event, param);
+    SuperWindowEvent(sender, event, param);
```
Comment (mkbel) [inline src/gui/screen_menu_sensor_info.cpp:22]: Please explain this change.

File: src/gui/screen_menu_odometer.cpp [hunk 3]
```diff
@@ -1,53 +1,29 @@
...
-    static constexpr const char *label = N_("ODOMETER");
-
-    static float getVal(Odometer_s::axis_t axis) {
-        Odometer_s::instance().force_to_eeprom();
-        return Odometer_s::instance().get(axis) * .001f;
-    }
-    static uint32_t getTime() {
-        Odometer_s::instance().force_to_eeprom();
-        return Odometer_s::instance().get_time();
-    }
```
Comment (mkbel) [inline src/gui/screen_menu_odometer.cpp:27]: `getVal` and `getTime` methods disappeared. Is this change intended?

File: src/gui/screen_prusa_link.cpp [hunk 4]
```diff
@@ -11,125 +12,46 @@
...
-class ScreenMenuPrusaLink : public AddSuperWindow<screen_t> {
-    constexpr static const char *const label = N_("PRUSA LINK");
-    static constexpr ResourceId canvas_font = IDR_FNT_SPECIAL;
+MI_PL_PASSWORD_LABEL::MI_PL_PASSWORD_LABEL()
+    : WI_LABEL_t(_(label), 0) {}
 
-    PLMenuContainer container;
-    window_menu_t menu;
-    window_header_t header;
+void MI_PL_PASSWORD_VALUE::MI_PL_PASSWORD_VALUE::printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const {
```
Comment (mkbel) [inline src/gui/screen_prusa_link.cpp:37]: `MI_PL_PASSWORD_VALUE::MI_PL_PASSWORD_VALUE::` Why is there MI_PL_PASSWORD_VALUE twice?

File: src/gui/screen_menu_experimental_settings_debug.hpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    int32_t microsteps_y;
+    int32_t microsteps_z;
+    int32_t microsteps_e;
+    int32_t rms_current_ma_x;
+    int32_t rms_current_ma_y;
+    int32_t rms_current_ma_z;
+    int32_t rms_current_ma_e;
+
+    // this is only safe as long as there are no gaps between variabes
+    // all variables re 32bit now, so it is safe
```
Comment (mkbel) [inline src/gui/screen_menu_experimental_settings_debug.hpp:38]: typo re -> image

File: src/gui/test/screen_menu_test.hpp [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+#include "screen_test_gui.hpp"
+#include "screen_test_term.hpp"
+#include "screen_test_msgbox.hpp"
+#include "screen_qr_error.hpp"
+#include "screen_test_wizard_icons.hpp"
+#include "screen_test_dlg.hpp"
+#include "screen_menu_eeprom_test.hpp"
+#include "screen_test_load.hpp"
+
+//#include "menu_opener.hpp" TODO make it work
```
Comment (mkbel) [inline src/gui/test/screen_menu_test.hpp:17]: TODO left

File: src/gui/test/screen_menu_test.hpp [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    MI_SELFTEST_TEST();
+
+protected:
+    virtual void click(IWindowMenu &window_menu) override;
+};
+
+using ScreenMenuTest__ = ScreenMenu<GuiDefaults::MenuFooter, MI_RETURN,
+
+    // GENERATE_SCREEN_FN_ITEM_DEV(GetScreenMenuEepromTest, "test EEPROM"),
+    /* TODO make it work
```
Comment (mkbel) [inline src/gui/test/screen_menu_test.hpp:55]: TODO left

File: src/gui/screen_menu_info.hpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/screen_menu_info.hpp:16]: it is not, I have missed it, thanks

File: src/gui/screen_menu_sensor_info.cpp (hunk 2 above)
Comment (RadekVana) [reply src/gui/screen_menu_sensor_info.cpp:22]: It is the same.
Screen:: was unnecessary
It was just specification of parent name which contains that method

File: src/gui/screen_menu_odometer.cpp (hunk 3 above)
Comment (RadekVana) [reply src/gui/screen_menu_odometer.cpp:27]: it is unused, even in private

File: src/gui/screen_prusa_link.cpp (hunk 4 above)
Comment (RadekVana) [reply src/gui/screen_prusa_link.cpp:37]: bug
I don't know why it compiles

File: src/gui/test/screen_menu_test.hpp (hunk 6 above)
Comment (RadekVana) [reply src/gui/test/screen_menu_test.hpp:17]: It currently does not work
I plan to fix it later
It is test menu, which is only in debug

File: src/gui/test/screen_menu_test.hpp (hunk 7 above)
Comment (RadekVana) [reply src/gui/test/screen_menu_test.hpp:55]: It currently does not work
I plan to fix it later
It is test menu, which is only in debug

File: src/gui/screen_menu_info.hpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/screen_menu_info.hpp:16]: fixed

File: src/gui/screen_menu_odometer.cpp (hunk 3 above)
Comment (RadekVana) [reply src/gui/screen_menu_odometer.cpp:27]: fixed

File: src/gui/screen_prusa_link.cpp (hunk 4 above)
Comment (RadekVana) [reply src/gui/screen_prusa_link.cpp:37]: fixed

File: src/gui/screen_menu_experimental_settings_debug.hpp (hunk 5 above)
Comment (RadekVana) [reply src/gui/screen_menu_experimental_settings_debug.hpp:38]: fixed

**Review by mkbel — APPROVED** [review]
